lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,161 @@
+const BN = require('../../../bn')
+
+const unitScale = 6
+
+class CardanoAPI {
+  constructor(client, port, hostname = 'localhost') {
+    this.client = client
+    this.port = port
+    this.hostname = hostname
+  }
+
+  async fetch(endpoint, params, data, appendData = []) {
+    const method = data ? 'POST' : 'GET'
+
+    const url = `https://${this.hostname}:${this.port}/api/v1/${endpoint}`
+
+    return Promise.resolve(true)
```

Comment:
Is this needed? Is not `this.client` async by default?

Reply:
:+1: